packages/models-library/src/models_library/api_schemas_webserver/wallets.py

```diff
@@ -37,3 +44,34 @@ class PutWalletBodyParams(OutputSchema):
     description: str | None
     thumbnail: str | None
     status: WalletStatus
+
+
+#
+# Payments to top-up credits in wallets
+#
+
+PaymentID: TypeAlias = IDStr  # identifier associated to a payment transaction
+
+
+class CreateWalletPayment(InputSchema):
+    price_dollars: Decimal
+    osparc_credits: Decimal  # NOTE: should I recompute? or should be in the backend?
+    comment: str = FieldNotRequired(max_length=100)
+
+
+class WalletPaymentCreated(OutputSchema):
+    payment_id: PaymentID
+    payment_form_url: HttpUrl = Field(
+        ..., description="Link to external site that holds the payment submission form"
+    )
+
+
+class PaymentTransaction(OutputSchema):
+    payment_id: PaymentID
+    price_dollars: Decimal
```

Comment:
Question: will this always be in dollars?

Reply:
yes. for the moment it is. We do not want to include currency right now to avoid increasing the logic complexity right now